app/src/main/java/com/findmyrecycling/service/ProductService.kt

```diff
@@ -13,9 +13,8 @@ class ProductService{
     suspend fun fetchProducts() : List<Product>? {
         return withContext(Dispatchers.IO) {
             val service = RetrofitClientInstance.retrofitInstance?.create(IProductDAO::class.java)
-            val products = async {service?.getAllProducts()}
-            var result = products.await()?.awaitResponse()?.body()
-            return@withContext result
+            val products = async { service?.getAllProducts() }
+            return@withContext products.await()?.awaitResponse()?.body()
```

Comment:
updated variable to inline

Reply:
I caution against doing too much in one line.  If you get an exception on this line, it's hard to determine what went wrong.